Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -377,7 +377,34 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         checker.visit(document)
         diagnosticEngine.emit(checker.problems)
     }
-
+    
+    /// A cache of plain string module names, keyed by the module node reference.
+    private var moduleNameCache: [ResolvedTopicReference: String] = [:]
```

Comment:
Not sure I understand the need for this cache, rather than just using the node's title. Is this for performance?

Reply:
Yes, this is only for performance. The module symbol's name will be displayed on all symbol's pages so it's a good candidate for a value to cache.

Reply:
Ok, that makes sense. Wondering if we should make this an IOU so that we don't accidentally access it before it's been populated. We do the same for `DocumentationContext.rootModules`.

Reply:
I think that's a good idea. I made this non-optional and added asserts about how this API is used.